kernel/src/transaction/create_table.rs

```diff
@@ -0,0 +1,991 @@
+//! Create table transaction implementation.
+//!
+//! This module provides a type-safe API for creating Delta tables.
+//! [`CreateTableTransactionBuilder`] configures table properties and builds a [`Transaction`]
+//! that can be committed to create the table.
+//!
+//! # Example
+//!
+//! ```rust,no_run
+//! use delta_kernel::table_manager::TableManager;
+//! use delta_kernel::schema::{StructType, StructField, DataType};
+//! use delta_kernel::expressions::ColumnName;
+//! use delta_kernel::committer::FileSystemCommitter;
+//! use delta_kernel::transaction::DataLayout;
+//! use std::sync::Arc;
+//! use std::collections::HashMap;
+//! # use delta_kernel::Engine;
+//! # fn example(engine: &dyn Engine) -> delta_kernel::DeltaResult<()> {
+//!
+//! let schema = Arc::new(StructType::try_new(vec![
+//!     StructField::new("id", DataType::INTEGER, false),
+//!     StructField::new("name", DataType::STRING, true),
+//! ])?);
+//!
+//! // Create a clustered table
+//! let result = TableManager::create_table("/path/to/table", schema.clone(), "MyApp/1.0")
+//!     .with_data_layout(DataLayout::Clustered(vec![ColumnName::new(["id"])]))
+//!     .build(engine, Box::new(FileSystemCommitter::new()))?
+//!     .commit(engine)?;
+//!
+//! // Or create a partitioned table
+//! let result = TableManager::create_table("/path/to/table2", schema, "MyApp/1.0")
+//!     .with_data_layout(DataLayout::Partitioned(vec!["id".to_string()]))
+//!     .build(engine, Box::new(FileSystemCommitter::new()))?
+//!     .commit(engine)?;
+//! # Ok(())
+//! # }
+//! ```
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use url::Url;
+
+use crate::actions::{
+    DomainMetadata, Metadata, Protocol, TABLE_FEATURES_MIN_READER_VERSION,
+    TABLE_FEATURES_MIN_WRITER_VERSION,
+};
+use crate::clustering::ClusteringMetadataDomain;
+use crate::committer::Committer;
+use crate::schema::{ColumnName, SchemaRef};
+use crate::table_features::{
+    assign_column_mapping_metadata, extract_feature_overrides,
+    get_column_mapping_mode_from_properties, ColumnMappingMode, TableFeature,
+    COLUMN_MAPPING_MAX_COLUMN_ID_KEY,
+};
+use crate::transaction::data_layout::{DataLayout, MAX_CLUSTERING_COLUMNS};
+use crate::transaction::Transaction;
+use crate::utils::{current_time_ms, try_parse_uri};
+use crate::{DeltaResult, Engine, Error};
+
+/// Table property key for specifying the minimum reader protocol version.
+const MIN_READER_VERSION_PROP: &str = "delta.minReaderVersion";
+/// Table property key for specifying the minimum writer protocol version.
+const MIN_WRITER_VERSION_PROP: &str = "delta.minWriterVersion";
+
+/// Result of extracting table features from properties.
+struct ExtractedFeatures {
```

Comment:
nit: `ExtractedFeatures` doesn't really sound like something that will hold `cleaned_properties` // these cleaned properties are not delta table features